src/lib/core/PeerId.h

```diff
@@ -39,15 +40,49 @@ class PeerId
         return *this;
     }
 
-    FabricId GetFabricId() const { return mFabricId; }
-    PeerId & SetFabricId(FabricId id)
+    CompressedFabricId GetCompressedFabricId() const { return mCompressedFabricId; }
+    PeerId & SetCompressedFabricId(CompressedFabricId id)
+    {
+        mCompressedFabricId = id;
+        return *this;
+    }
+
+    bool operator==(const PeerId & other) const
+    {
+        return (mNodeId == other.mNodeId) && (mCompressedFabricId == other.mCompressedFabricId);
+    }
+    bool operator!=(const PeerId & other) const
+    {
+        return (mNodeId != other.mNodeId) || (mCompressedFabricId != other.mCompressedFabricId);
+    }
+
+private:
+    NodeId mNodeId = kUndefinedNodeId;
+
+    CompressedFabricId mCompressedFabricId = kUndefinedFabricId;
+};
+
+class UncompressedPeerId
+{
+public:
+    UncompressedPeerId() {}
+
+    NodeId GetNodeId() const { return mNodeId; }
+    UncompressedPeerId & SetNodeId(NodeId id)
+    {
+        mNodeId = id;
+        return *this;
+    }
+
+    FabricId GetUncompressedFabricId() const { return mFabricId; }
+    UncompressedPeerId & SetUncompressedFabricId(FabricId id)
     {
         mFabricId = id;
         return *this;
     }
 
-    bool operator==(const PeerId & other) const { return (mNodeId == other.mNodeId) && (mFabricId == other.mFabricId); }
-    bool operator!=(const PeerId & other) const { return (mNodeId != other.mNodeId) || (mFabricId != other.mFabricId); }
+    bool operator==(const UncompressedPeerId & other) const { return (mNodeId == other.mNodeId) && (mFabricId == other.mFabricId); }
```

Comment:
This also need to account for root, as previously mentioned.

Reply:
https://github.com/project-chip/connectedhomeip/issues/9192